File: backend/app/schemas/product.py
from uuid import UUID

from pydantic import BaseModel


class ProductResponse(BaseModel):
    id: UUID
    name: str
    sku: str
    description: str | None = None
    unit: str
    is_available: bool
    category_id: UUID
    
    
class ProductListResponse(BaseModel):
    items: list[ProductResponse]
    page: int
    page_size: int
    total: int
    total_pages: int

File: frontend/types/product.ts
export interface Product {
  id: string;
  name: string;
  sku: string;
  description: string | null;
  unit: string;
  stock_quantity: number;
  is_available: boolean;
  category_id: string;
}

export interface ProductsResponse {
  items: Product[];
  page: number;
  page_size: number;
  total: number;
  total_pages: number;
}

export interface GetProductsParams {
  search?: string;
  category_id?: string;
  page?: number;
  page_size?: number;
}